crates/runtime/src/traphandlers/backtrace.rs

```diff
@@ -155,6 +196,63 @@ impl Backtrace {
         log::trace!("====== Done Capturing Backtrace (reached end of activations) ======");
     }
 
+    unsafe fn trace_through_continuations(
+        chain: Option<&StackChain>,
+        pc: usize,
+        fp: usize,
+        trampoline_sp: usize,
+        mut f: impl FnMut(Frame) -> ControlFlow<()>,
+    ) -> ControlFlow<()> {
+        // Handle the stack that is currently running (which may be a
+        // continuation or the main stack).
+        Self::trace_through_wasm(pc, fp, trampoline_sp, &mut f)?;
+
+        chain.map_or(ControlFlow::Continue(()), |chain| {
+            debug_assert_ne!(*chain, StackChain::Absent);
+
+            let stack_limits_iter = chain.clone().into_iter();
+
+            // The very first entry in the stack chain is for what is currently
+            // running (which may be a continuation or main stack). However, for
+            // the currently running stack, the data in its associated
+            // `StackLimits` object is stale (see comment on
+            // `wasmtime_continuations::StackChain` for a description of the
+            // invariants).
+            // That's why we already handled the currently running stack at the
+            // beginning of the function, using data directly from the
+            // VMRuntimeLimits.
+            let remainder = stack_limits_iter.skip(1);
+
+            for (continuation_opt, limits) in remainder {
+                let limits = limits.as_ref().unwrap();
+                match continuation_opt {
+                    Some(continuation) => {
+                        let cont = unsafe { &*continuation };
```

Comment:
Why not simply use an `as` cast here? The dereference performs a move

The unsafe block annotation should arguably be around the entire match arm, because the `&*continuation` could violate the strict aliasing rules. We should add a comment saying that this line is the unsafe bit of the arm.

Reply:
I'm pretty sure that making a piece of code `unsafe` doesn't change the permitted aliasing. The aliasing here should be fine: We aren't violating the rule that there must be no more than one mutable reference at a time (because there are zero mutable references here).

Reply:
That's not what I am saying. The entire block would be unsafe if we were to alias the continuation. The safety of the rest of code relies on this assumption